File: src/java/com/afollestad/polarupgradetool/jfx/InterfaceUpdateThread.java
package com.afollestad.polarupgradetool.jfx;

import com.afollestad.polarupgradetool.Main;

/**
 * Project : polar-dashboard-upgrade-tool
 * Author : pddstudio
 * Year : 2016
 */
public class InterfaceUpdateThread extends Thread implements Runnable {

    final String location;
    final UICallback uiCallback;

    public InterfaceUpdateThread(String location, UICallback uiCallback) {
        this.location = location;
        this.uiCallback = uiCallback;
    }

    @Override
    public void run() {
        Main.upgrade(location, uiCallback);
    }

}

File: src/main/java/com/afollestad/polarupgradetool/jfx/InterfaceUpdateThread.kt
package com.afollestad.polarupgradetool.jfx

import com.afollestad.polarupgradetool.Main

/**
 * Project : polar-dashboard-upgrade-tool
 * Author : pddstudio
 * Year : 2016
 */
class InterfaceUpdateThread(internal val location: String, internal val uiCallback: UICallback) : Thread(), Runnable {

    override fun run() {
        Main.upgrade(location, uiCallback)
    }
}